### ISSUE TITLE
entry/exit signals vs trades

### ISSUE THREAD

**Issue by darioneto** [freqtrade/freqtrade#8177]
## Describe your environment

  * Operating system: __Docker__
  * Python Version: __Python 3.10.7___ (`python -V`)
  * CCXT version: __ccxt==2.7.12___ (`pip freeze | grep ccxt`)
  * Freqtrade Version: __freqtrade docker-2023.2.dev-a7fec1f8__ (`freqtrade -V` or `docker compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## Your question

the bot seems to generate entry/exit signals (during backtesting) but during the live run doesn't take trades on some of them. Not sure what this could be related to. Is this expected? 

![image](https://user-images.githubusercontent.com/31065537/219238027-22c12e6a-30c4-4c05-9ebe-7b68c6594d83.png)

The second question is it possible to send a telegram notification every time the extry/exit signal was generated? 
I tried adding those below but this doesn't make any difference :

```
    allow_custom_messages: true
    notification_settings:
      status: "on"
      warning: "on"
      startup: "on"
      entry: "on"
      entry_fill: "on"
      entry_cancel: "on"
      exit:
        roi: "on"
        emergency_exit: "on"
        force_exit: "on"
        exit_signal: "on"
        trailing_stop_loss: "silent"
        stop_loss: "on"
        stoploss_on_exchange: "on"
        custom_exit: "on"
        partial_exit: "on"
      exit_cancel: "on"
      exit_fill: "on"
```

**Comment by stash86** [comment]
Two possibilities
1. Lookahead bias, hence backtest showing more trades than dry/live
2. Conflicting signals

You can use send_msg feature for that. You can't just add random events to config and hope it suddenly exists

**Comment by darioneto** [comment]
Ok, I see, do you know if I can find a working example with "send_msg" including sending each entry/exit signal? 
I tried to add this " self.dp.send_msg(f"{metadata['pair']} just got hot!")" under the populate_indicators section, but not sure how this is meant to be integrated with the entry/exit signal.

Are those signal messages being logged to the database by any chance so they can be fetched from the database or via API?

**Comment by xmatthias** [comment]
What you're seeing in the screenshot are only signals. There's one trade in the whole screenshot.
Trade signals will be ignored in the following cases:
* no open trade-slot is available (max_open_trades is reached)
* the pair already has an open trade
* signals are colliding
* lookahead bias (backtesting shows 1000nds of trades, but live shows 0 trades, or signas that "pop in" after 2-3 candles).

> Are those signal messages being logged to the database by any chance so they can be fetched from the database or via API?

messages are not logged to the database. it's a notification that's only really relevant when it happens.

**Comment by darioneto** [comment]
Thanks, I think colliding signals could be the reason for this. Just wondering why this would be TBC.. 
By the way, do you know if I can find a working example with "send_msg" including sending each entry/exit signal? It can be useful regardless of whether trades take place or not.

**Comment by xmatthias** [comment]
UI can help you with colliding signals - simply disable the entry signal - and the exit signals would show up.

> By the way, do you know if I can find a working example with "send_msg" including sending each entry/exit signal

Probably not. That'd be pointlessly verbose.

**Comment by darioneto** [comment]
yeah, you are right, that was caused by colliding signals just wondering why hyperopt resulted in colliding parameters. Stil produced positive results which are shown after backtesting and live, however, there are a lot of opposite signals at the same time. strange.
By the way, the signals are being shown in the UI only on the last 20 candles or so. Can this be changed?

**Comment by xmatthias** [comment]
Hyperopt is really just trying parameters to improve the loss function. results will differ depending on the function used.

Ui's will show signals where they appear in the strategy - if they're just on the last 20 candles - then the strategy will not have generated signals earlier than that.

**Comment by darioneto** [comment]
That is strange, once the chart is refreshed older signals disappear and they move forward to the latest candle so the constant number of past signals is 22
![image](https://user-images.githubusercontent.com/31065537/219318493-8f1676df-8c78-4060-951c-bc143cbbfb03.png)

**Comment by xmatthias** [comment]
Well - that's a pretty clear sign of one of the [common mistakes in strategy development](https://www.freqtrade.io/en/stable/strategy-customization/#common-mistakes-when-developing-strategies).

**Comment by darioneto** [comment]
Does it mean that those signals are looking certain period into the future? And that's why only appear 20 first candles? 
Didn't have enough time to check in details as this may probably going to take days to find out what is the problem. It seems that trades are being taken around the same period as the backtest results. Hard to tell what is the problem with this.

**Comment by stash86** [comment]
well, without looking at the strategy code, we can only point you to the article. You would have to debug it yourself

**Comment by darioneto** [comment]
my strategy is not that complex however derived from the internal DB. The hyperfunction is set to optimize a few formulas. The challenge is to determine whether this is actually a case of forward lookup here

```
buy_params = {
  "round_down_diff_threshold": 0.37,
  "round_down_pct_change_threshold": 0.1,
  "round_down_up_ratio_threshold": 1.246,
  "round_up_diff_threshold": -0.629,
}

sell_params = {
  "round_down_pct_change_threshold_exit": 2.0,
  "round_down_up_ratio_threshold_exit": 0.0793,
  "round_up_pct_change_threshold_exit": -2.5,
}

populate_indicators
dataframe['round_down_up_ratio'] = dataframe['round_down_5m'] / dataframe['round_up_5m']
dataframe['round_down_pct_change'] = dataframe['round_down_5m'].pct_change()
dataframe['round_down_diff'] = dataframe['round_down_5m'].diff()
dataframe['round_up_diff'] = dataframe['round_up_5m'].diff()
dataframe['round_up_pct_change'] = dataframe['round_up_5m'].pct_change()

Entry
conditions.append (
(dataframe['round_up_diff'] > self.round_up_diff_threshold.value) &
(dataframe['round_down_diff'] < self.round_down_diff_threshold.value) &
(dataframe['round_down_pct_change'] > self.round_down_pct_change_threshold.value) &
(dataframe['round_down_up_ratio'] < self.round_down_up_ratio_threshold.value) 
)

Exits
conditions.append(
(dataframe['round_up_pct_change'] > self.round_up_pct_change_threshold_exit.value) &
(dataframe['round_down_pct_change'] < self.round_down_pct_change_threshold_exit.value) &
(dataframe['round_down_up_ratio'] < self.round_down_up_ratio_threshold_exit.value) 
)
```

**Comment by stash86** [comment]
Well, the question is whether the round_down and round_up have lookahead or not

**Comment by darioneto** [comment]
round_up and round_down are real-time values aligned with dataframe. That's been working for a long time

```
        inf_tf = '5m'
        dataframe = merge_informative_pair(dataframe, dataframe_up, self.timeframe, inf_tf, ffill=True)
        dataframe = merge_informative_pair(dataframe, dataframe_down, self.timeframe, inf_tf, ffill=True)
```

**Comment by darioneto** [comment]
ok looks as if the problem could have been related to the overlapping signals or the collisions. What I did I added an additional condition to the exit to get rid of so many contradictive signals and that seems to fix it as far as I can see on the UI. All the extry signals are listed there for all candles as I'd expect to see initially. Not only for the last 20. 

```
            conditions.append(
                qtpylib.crossed_below(
                    dataframe[self.sell_fast_key_exit.value].shift(self.sell_horizontal_push_exit.value),
                    dataframe[self.sell_slow_key_exit.value] * self.sell_vertical_push_exit.value
                    ) 
            )
```
![image](https://user-images.githubusercontent.com/31065537/219376452-07cac06b-c452-487b-9662-05b4a6d710b1.png)
